### DISCUSSION TITLE
What's Next?

### DISCUSSION THREAD

**Discussion by robotlightsyou** [Slow-Scallion4183/Adventure-Colussus#20] [Ideas]
I've added what feels like a lot without getting a lot of input from OP. On top of that, in the original reddit post they said they were interested in keeping the driving force a little younger. I'm not young, but this is my first project to contribute to that isn't my own. I'd like to stay a part of this, but I'd also like y'all to lead the charge since that was OP's request. With that in mind, here's some things I'm thinking about that might be good next tasks:

Story ideas - we can start on basics mechanics and assets, but until there's an idea of the setting and length of game it might be hard to get too far. I don't think we need to settle on everything, we don't want to paint into a corner, but something a little more than zombies and loot might be good.

How does loot work? Are there stores? Is there money or all barter? Do NPCs join your group and give you access to their loot or do you only encounter them on the road?

Combat: I've built some mechanics for how an attack might be modified by weapon or luck, but they're very basic and not well thought out, lots of room to expand.

Levels and xp: I put in placeholders, but I haven't started implementing it at all.

Window: it might be a good idea to look into libraries that would set the window the same every time so we didn't have to worry about the ascii art displaying incorrectly.

This got long fast and it's pushing midnight here. I hope this gives y'all some things to think about, and maybe some ideas on what direction to go. Let me know if there's anything else you'd like me to be looking at, or if I'm overstepping. Have a good one.

**Comment by ghost** [comment]
For Window Do tkinter or easygui, just a expriment. he wants it to look like retro. make it like there are some chests around the spawnpoint of the player where theres loot which are mentioned in the ideas text file or treasure chests in the storyline. combat is good lol, i just checked. and armor is must. levels and xp is great for implementing skills like piercing attack in sword which does more damage but takes a bit of xp. story ideas will be discussed when these mechanics are finished and we can take ideas from other games like flare rpg and other games in github etc.

**Reply by Slow-Scallion4183** [reply]
whaaat no no no i meant text-based retro not graphics retro

**Reply by ghost** [reply]
ok lol

**Comment by Slow-Scallion4183** [comment]
Thanks for that robotlightsyou! Everyone is welcome to work on the project if they are from beginner to intermediate and so on. Its been great having you here and contributing so you can stay here and carry on no problem!!! 

For the story i was thinking 3 parts (beginning, middle and end obviously) but maybe something to go along the lines of:
    - Find a special object with the help of an npc, fight enemys along the way etc, find loot along the way
    -  Once you have that object you travel to an inn for example to meet another npc who will send you on a quest to find another special 
        object. Fight enemys along the way, find loot etc. 
    - When you have these two objects you need to go to some hidden, secretive place and combine these two objects and then make 
       your way to a boss fight (somewhere further away) and use this new found object to defeat him (on the way you fight enemys, find 
      loot) and you get special drops from fighting the randomised boss fight (eg dragon, mega troll etc)

Loot will work with 5 text files: ranged (ranged weapons), melee weapons (swords etc), items (health potions, 4 leaf clover for luck etc), shield (obsidian shield etc) and armor (wooden armor etc (will increase max health or something)). 

There will be a randomiser to select one of these for text files when you defeat an enemy, and they will drop one of these pieces of loot. This loot will then increase the corresponding stat (eg wooden bow + 10 ranged). 

Structure of the text file:
`wooden bow, 10` 
etc

The method we can use to extract this data is either with csv:
```import csv

with open('file_test.txt') as csvfile:
    reader = csv.reader(csvfile)
    for row in reader:
        var1, var2 = map(lambda s: s.strip(), row)

print(var1, var2)

```

or like so: 

```
with open("file_test.txt") as fh:  # fh is now a file-like object
    for line in fh:  # text file-likes are iterable by-line!
        var1, var2 = line.split(",")
# file is now closed when you leave the with scope!

print(var1, var2)
```

**Reply by robotlightsyou** [reply]
Cheers for the thorough follow up, there's a lot to chew on here. I'm gonna throw this in the hopper and think it over for a bit and get back to the group when I have some better formulated ideas.

**Reply by Slow-Scallion4183** [reply]
Nice! Sounds good! :D
